couchbase.go

```diff
@@ -123,22 +122,15 @@ func (c *CouchbaseDB) UpdateUser(ctx context.Context, req dbplugin.UpdateUserReq
 }
 
 func (c *CouchbaseDB) DeleteUser(ctx context.Context, req dbplugin.DeleteUserRequest) (dbplugin.DeleteUserResponse, error) {
-	c.Lock()
-	defer c.Unlock()
+	// Don't let anyone write the config while we're using it
+	c.RLock()
+	defer c.RUnlock()
 
 	db, err := c.getConnection(ctx)
 	if err != nil {
 		return dbplugin.DeleteUserResponse{}, fmt.Errorf("failed to make connection: %w", err)
 	}
 
-	// Close the database connection to ensure no new connections come in
-	defer func() {
-		if err := c.close(); err != nil {
-			logger := hclog.New(&hclog.LoggerOptions{})
-			logger.Error("defer close failed", "error", err)
-		}
-	}()
-
```

Comment:
This removal seems out of place and I'm wondering why this needed to be removed?

Reply:
If I understand it correctly, this logic only would apply to root credential rotation but the DB engine closes the plugin on a rotate-root call so this is unnecessary (and actually problematic for static credential rotation). Can you confirm this is correct or am I mistaken @fairclothjm ?

Reply:
Okay, thanks!